### DISCUSSION TITLE
Detect mobile devices in server components

### DISCUSSION THREAD

**Discussion by mdjfs** [vercel/next.js#58863] [Help]
### Summary

i need switch content of an app between mobile devices and desktop mobiles. In client components. I just use Web APIs to detect user device. But it is slowest. Any way to detect in react components the device of the user? Like the fastest way to switch the page structure

**Comment by hereisfahad** [comment]
Did you find any solution to this? Maybe try using this: https://nextjs.org/docs/app/api-reference/functions/userAgent.

**Comment by SouthLink** [comment]
I created a function to solve this problem

```
import { headers } from "next/headers";
import { userAgentFromString } from "next/server";

/**
 * @description next.js mobile detection
 */
export function isServerMobile() {
  const headersList = headers();
  const userAgent = headersList.get("user-agent");
  const { device } = userAgentFromString(userAgent || undefined);

  return device.type === "mobile";
}

```
